### PR TITLE
Add `exports` field in package.json

### PR DESCRIPTION
**Pull request by gdiazdelaserna**
### TL;DR

Fixes the issue mentioned here: https://github.com/JedWatson/react-select/issues/4859

#### Issue
Consuming a React Select subpath from a JS module in a project using Webpack 5, throws an error saying the path in the  import statement is not fully specified.

#### Context
Webpack 5 introduced a default behavior that requires all import statements in JS modules to be [fully specified](https://webpack.js.org/configuration/module/#resolvefullyspecified).

Importing a React Select subpath (e.g. `import Creatable from 'react-select/creatable'`) is breaking that rule. Webpack 5 thinks that import statement is trying to import a file with a missing extension.

#### Solution
Inform Webpack that the subpaths are not file imports (and an extension shouldn't be required) by declaring an `exports` field in `package.json`.

The `exports` field is documented in these two places:
- [Node](https://nodejs.org/api/packages.html#exports)
- [Webpack](https://webpack.js.org/guides/package-exports/)

The new `exports` field includes all the subpaths mentioned in React Select's documentation. All other subpath imports will fail from now on.

Each subpath declared in the `exports` field includes 3 different file system paths to satisfy the different environments that might consume them:
- `require`: Will be used by environments using the `require` keyword to import files (e.g CommonJS)
- `import`: Will be used by environments using the `import` keyword to import files (e.g. ES Modules)
- `types`: Will be used by TypeScript to query type definitions

### CODE REVIEW

**Comment by Rall3n** [comment]
@gdiazdelaserna Has this been tested with webpack and vite?

**Comment by gdiazdelaserna** [comment]
@Rall3n 

I have tested this change in multiple environments. I created this repo to show how this has been tested: 

https://github.com/gdiazdelaserna/react-select-exports-field-test

As you can see, I tested this in:
- Create React App
- Next.js
- Vite
- Webpack

This proves that:
- React Select continues to work correctly in all those environments (i.e. there are no regressions)
- The PR fixes the issue it intends to fix: React Select subpaths imports now work correctly when consumed from a JS module in a Webpack 5 app

**Review by Rall3n — APPROVED** [review]

**Comment by davbrito** [comment]
As a workaround I put the path to the file on node_modules

image
